HeterogeneousCore/AlpakaInterface/interface/atomicIncSaturate.h

```diff
@@ -0,0 +1,29 @@
+#ifndef HeterogeneousCore_AlpakaInterface_interface_atomicIncSaturate_h
+#define HeterogeneousCore_AlpakaInterface_interface_atomicIncSaturate_h
+
+#include <alpaka/alpaka.hpp>
+
+// This function is similar to atomicInc, but instead of wrapping around it saturates at the given value.
+
+ALPAKA_NO_HOST_ACC_WARNING
+template <typename TAcc, typename T, typename THierarchy = alpaka::hierarchy::Grids>
+ALPAKA_FN_HOST_ACC auto atomicIncSaturate(TAcc const& acc,
+                                          T* address,
+                                          T const& limit,
+                                          THierarchy const& hierarchy = THierarchy()) -> T {
+  T assumed;
+  T old = *address;
```

Comment:
On host parallel backends (that we don't use at the moment) this line is not thread safe (actually, strictly speaking with the default `Grids` hierarchy level this line is not thread safe even with host serial backend if the same `address` is accessed concurrently from multiple kernels). To guarantee thread safety on host the load from `address` would have to be atomic.

Reply:
I see.
I will have to think how it can be implemented in a portable way, or specialised for the various backends.